dds/DCPS/XTypes/DynamicData.cpp

```diff
@@ -2082,9 +2098,21 @@ bool DynamicData::skip_collection_member(TypeKind kind)
     }
     return false;
   }
-
-  const DCPS::String err_msg = DCPS::String("Failed to skip a non-primitive ") + kind_str + " member";
-  return skip("skip_collection_member", err_msg.c_str(), dheader);
+  if (strm_.encoding().kind() == DCPS::Encoding::KIND_XCDR2) {
+    const DCPS::String err_msg = DCPS::String("Failed to skip a non-primitive ") + kind_str + " member";
+    return skip("skip_collection_member", err_msg.c_str(), dheader);
+  } else if (kind == TK_SEQUENCE) {
+    return skip_to_sequence_element(0, coll_type, true);
+  } else if (kind == TK_ARRAY) {
+    return skip_to_array_element(0, coll_type, true);
+  } else if (kind == TK_MAP) {
+    if (DCPS::log_level >= DCPS::LogLevel::Error) {
+      ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: DynamicData::skip_collection_member: "
+                 "Maps are not currently supported in OpenDDS\n"));
```

Comment:
This probably shouldn't be an error, since errors in the new logging scheme are very serious and should only be logged when OpenDDS is more or less not able to function.